git-tools.js

```diff
@@ -187,6 +187,44 @@ Repo.prototype.authors = function( committish, callback ) {
 	});
 };
 
+Repo.prototype.describe = function( options, callback ) {
+	if ( !callback ) {
+		callback = options;
+		options = { };
+	}
+
+	var args = [ "describe" ];
+	if( options.all ) {
+		args.push( "--all" );
+	}
+	if( options.tags ) {
+		args.push( "--tags" );
+	}
+	if( options.contains ) {
+		args.push( "--contains" );
+	}
+	if( options.debug ) {
+		args.push( "--debug" );
+	}
+	if( options.long ) {
+		args.push( "--long" );
+	}
+	if( options.always ) {
+		args.push( "--always" );
+	}
+	if( options.first_parent ) {
+		args.push( "--first-parent" );
+	}
```

Comment:
My thought was that by spelling it out like this makes sure only supported parameters are passed to git. But using your technique will give back an error anyway if they supply wrong parameters, I guess I didn't think of that..